modules/custom/az_demo/tests/src/Functional/AZDemoContentTest.php

```diff
@@ -87,6 +87,16 @@ public function testPublicationLinks() {
     $assert = $this->assertSession();
     // Assert the page loads successfully.
     $assert->statusCodeEquals(200);
+    // Assert individual links exist with the correct text and href.
+    $assert->elementExists('xpath', "//a[@href='/publication/life-leonardo-da-vinci-0' and text()='The Life of Leonardo Da Vinci']");
+    $assert->elementExists('xpath', "//a[@href='https://example.com/wildcat-book-review' and text()='The Life of Leonardo Da Vinci']");
+    $assert->elementExists('xpath', "//a[@href='/publication/most-fearsome-life-great-gargantua-father-pantagruel' and text()='The Most Fearsome Life of the Great Gargantua, Father of Pantagruel.']");
+    $assert->elementExists('xpath', "//a[@href='https://example.com/data-structures-book' and text()='Advanced Data Structures']");
+    $assert->elementExists('xpath', "//a[@href='https://example.com/machine-learning-paper' and text()='Modern Approaches to Machine Learning']");
+    $assert->elementExists('xpath', "//a[@href='https://example.com/algorithms-book' and text()='Introduction to Algorithms']");
+    $assert->elementExists('xpath', "//a[@href='https://example.com/deep-learning-book' and text()='Deep Learning']");
+    $assert->elementExists('xpath', "//a[@href='https://example.com/climate-change-journal' and text()='Climate Change: Impacts and Solutions']");
+    $assert->elementExists('xpath', "//a[@href='/sites/default/files/Small-PDF.pdf' and text()='The Trissotetras: Or, a Most Exquisite Table for Resolving All Manner of triangles. ']");
```

Comment:
The actual path to the files folder seems to be a generated site for the test case, rather than `default`:
```
<div class="csl-entry"><a href=/person/thomas-urquhart>Sir Thomas Urquhart</a>. "<a href=/sites/simpletest/11415319/files/Small-PDF.pdf>The Trissotetras: Or, a Most Exquisite Table for Resolving All Manner of triangles. </a>". <i>Book</i>, 2nd Edition, Mathematical Works Publishing, 15 Jan. 2023, p. 400, <a href="https://doi.org/10.1000/2000">https://doi.org/10.1000/2000</a>. <a href="https://example.com/trissotetras">https://example.com/trissotetras</a></div>
```

```suggestion
    $siteDirectory = $this->siteDirectory;
    $assert->elementExists('xpath', "//a[@href='/{$siteDirectory}/files/Small-PDF.pdf' and text()='The Trissotetras: Or, a Most Exquisite Table for Resolving All Manner of triangles. ']");
```